### ISSUE TITLE
I DON'T KNOW CODES

### ISSUE THREAD

**Issue by yirki** [yirki/python3-machine-learning#1]
import numpy as np
import matplotlib.pyplot as plt
import sklearn.datasets as ds

lris=ds.load_iris()
xdata=lris.data.astype(np.float32)
tdata=lris.data.astype(np.int32)

print(tdata)
#Dtrain== 训练data D-Dtrain==test data

D,N=xdata.shape

Dtrain=D//2
index=np.random.permutation(range(D))
xtrain=xdata[index[0:Dtrain],:]
ttrain=tdata[index[0:Dtrain]]
xtest=xdata[index[Dtrain:D],:]
ttest=tdata[index[Dtrain:D]]

import chainer.optimizers as Opt
import chainer.functions as F
import chainer.links as L
from chainer import Variable,Chain,config

C=np.max(tdata)+1
NN=Chain(l1=L.Linear(N,3),l2=L.Linear(3,3),l3=L.Linear(3,C))

def model(x):
  h=NN.l1(x)
  h=F.sigmoid(h) #机器学习中的一个常用函数
  h=NN.l2(h)
  h=F.sigmoid(h)
  y=NN.l3(h)
  return y

optNN=Opt.SGD()
optNN.setup(NN)

train_loss=[]
train_acc=[]
test_loss=[]
test_acc=[]

#需要搞一个结界
T=1000
for time in range(T):
  config.train=True
  optNN.target.zerograds()
  ytrain=model(xtrain)
  **loss_train=F.softmax_cross_entropy(ytrain,ttrain)** #What's wrong with this raw?
  acc_train=F.accuracy(ytrain,ttrain)
  loss_train.backward()
  optNN.update()
  config.train=False
  ytest=model(xtest)
  loss_test=F.softmax_cross_entropy(ytest,ttest)
  acc_test=F.accuracy(ytest,ttest)
  train_loss.append(loss_train.data)
  train_acc.append(acc_train.data)
  test_loss.append(test_loss.data)
  test_acc.append(test_acc,data)

Tall=len(train_loss)
plt.figure(figsize=(8,6))
plt.plot(range(Tall),train_loss)
plt.plot(range(Tall),train_acc)
plt.title('loss function in training')
plt.xlabel('step')
plt.ylabel('loss function')
plt.xlim([0,Tall])
plt.ylim([0,4])
plt.show()
